core/src/main/java/hudson/model/Actionable.java

```diff
@@ -73,13 +74,23 @@
      */
     @Deprecated
 	public List<Action> getActions() {
-		if(actions == null) {
-			synchronized (this) {
-				if(actions == null) {
-					actions = new CopyOnWriteArrayList<Action>();
-				}
-			}
-		}
+        if(actions == null) {
+            synchronized (this) {
+                if (actions == null) {
+                    actions = new CopyOnWriteArrayList<Action>();
+                } else {
+                    Iterator actionIterator = actions.iterator();
```

Comment:
Or better, use `Iterator<Object>` and check not `action == null` but `!(action instanceof Action)`, so you preclude `ClassCastException`s from `this.actions` too.